iroh/src/endpoint.rs

```diff
@@ -1539,38 +1530,13 @@ impl Connection {
         self.inner.congestion_state()
     }
 
-    /// Parameters negotiated during the handshake.
-    ///
-    /// Guaranteed to return `Some` on fully established connections or after
-    /// [`Connecting::handshake_data()`] succeeds. See that method's documentations for
-    /// details on the returned value.
-    ///
-    /// [`Connection::handshake_data()`]: crate::Connecting::handshake_data
-    #[inline]
-    pub fn handshake_data(&self) -> Option<Box<dyn Any>> {
-        self.inner.handshake_data()
-    }
-
     /// Extracts the ALPN protocol from the peer's handshake data.
-    pub fn alpn(&self) -> Option<Vec<u8>> {
-        let data = self.handshake_data()?;
-        match data.downcast::<quinn::crypto::rustls::HandshakeData>() {
-            Ok(data) => data.protocol,
-            Err(_) => None,
-        }
-    }
-
-    /// Cryptographic identity of the peer.
-    ///
-    /// The dynamic type returned is determined by the configured [`Session`]. For the
-    /// default `rustls` session, the return value can be [`downcast`] to a
-    /// <code>Vec<[rustls::pki_types::CertificateDer]></code>
-    ///
-    /// [`Session`]: quinn_proto::crypto::Session
-    /// [`downcast`]: Box::downcast
-    #[inline]
-    pub fn peer_identity(&self) -> Option<Box<dyn Any>> {
-        self.inner.peer_identity()
+    pub fn alpn(&self) -> Vec<u8> {
+        let data = self.inner.handshake_data().expect("iroh crypto setup");
+        let data = data
+            .downcast::<quinn::crypto::rustls::HandshakeData>()
+            .expect("iroh crypto setup");
+        data.protocol.expect("ALPN required by iroh crypto setup")
```

Comment:
> Guaranteed to be set if a nonempty list of protocols was specified for this connection.

Are we sure this will always be the case from the public api usage?